plugins/kubernetes-backend/src/cluster-locator/CatalogClusterLocator.ts

```diff
@@ -64,14 +57,9 @@ export class CatalogClusterLocator implements KubernetesClustersSupplier {
       const clusterDetails: ClusterDetails = {
         name: entity.metadata.name,
```

Comment:
An open question (and potentially out of scope of the changes in this PR), but one of the core problems with using the Catalog Cluster Locator is this hardcoding of `entity.metadata.name`.

We have _many_ EKS clusters that we're auto ingesting from AWS through some custom processors, but we duplicate names across AWS accounts (to represent the same cluster across environments/regions), so we have to change the `entity.metadata.name` to be unique (I know we're off of the paved road here).

We are using `metadata.title` on the Entity to represent the actual name for how our developers know the EKS clusters. This seems to be a common use case based on https://github.com/backstage/backstage/issues/13793#issuecomment-1699749223

These changes as is would help us do proper auth mapping, but we can't really get the proper naming very easily (hardcoding a mapping of name to title is out of the question). 

Is the preferred approach that the auth strategy be the proxy for setting titles/display names for clusters or is there room to have "title" support in locators?

Reply:
I'm hoping for something along the lines of "title" support in locators -- I've got it in my todo list to expand that linked comment into its own issue where we can capture the implications for locators.

Reply:
@alecjacobs5401 I finally got around to writing up #21398